dartagnan/src/main/java/com/dat3m/dartagnan/verification/solving/RefinementSolver.java

```diff
@@ -605,13 +608,54 @@ private boolean process(Constraint c) {
     }
 
     /*
-        This pass rewrites the memory model so that no positive constraints appear on the rhs-side of a difference.
-        Suppose "r = a \ b" where r and a are negative and b is positive.
+        This pass rewrites the memory model so that no non-negative constraints appear on the RHS of a difference.
+        Consider "r = a \ b" where b is non-negative (note that r and a must be non-positive then).
         For example, this is the case if there is a constraint "~empty(r)"
         We rewrite the model to
-            let b`= free()
+            let b'= free()
             let r = a \ b' // b' gets cut, but it is effectively a base relation so this causes no large encoding.
-            empty (r & b) // b is now in a positive position and thus not need to get cut
+            empty (r & b) // b is now in a positive position and thus not need to get cut.
+
+        We claim that this model is equisatisfiable to the original memory model.
+        To understand this intuitively, it is helpful so think of the fresh b' as the original b but with "some slack".
+        We will argue that the best choice for b' is precisely b.
+        Any larger choice for b' will make r smaller, but since r is non-positive this reduces the chances
+        of satisfying negated axioms (e.g., if "~empty(r)" is true for some small r, then it is also true for all larger r).
+        On the other hand, if b' is chosen too small, then r gets so large that it violates "empty(r & b)".
```

Comment:
I had completely missed the fact that you also transformed the axiom. Now it makes much more sense

Reply:
The original axioms are still there. I only add a new axiom.

Reply:
Is this true: "the flagged axiom is still there, it is just that caat ignores it"

Reply:
Since the flagged axiom is negative, it also gets cut by the cutting algorithm. So CAAT doesn't know that it even exists.